hs_file_types/forms.py

```diff
@@ -0,0 +1,282 @@
+import json
+import jsonschema
+from urllib.request import Request, urlopen
+from urllib.error import URLError, HTTPError
+
+from django import forms
+from .models.model_program import ModelProgramResourceFileType
+from .utils import get_model_program_aggregations
+
+
+class ModelInstanceMetadataValidationForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user', None)
+        super(ModelInstanceMetadataValidationForm, self).__init__(*args, **kwargs)
+
+    has_model_output = forms.BooleanField(required=False)
+    executed_by = forms.IntegerField(required=False)
+    executed_by_url = forms.URLField(required=False)
+
+    def clean_executed_by(self):
+        executed_by = self.cleaned_data['executed_by']
+        user_selected_mp_aggr = None
+        # if a model program has been selected then this form would have the id of that aggregation
+        if executed_by is not None and executed_by > 0:
+            user_mp_aggregations = get_model_program_aggregations(self.user)
+            user_selected_mp_aggr = [mp_aggr for mp_aggr in user_mp_aggregations if mp_aggr.id == executed_by]
+            if user_selected_mp_aggr:
+                user_selected_mp_aggr = user_selected_mp_aggr[0]
+            else:
+                self.add_error("executed_by", "You don't have access to the selected model program aggregation")
+        return user_selected_mp_aggr
+
+    def clean_executed_by_url(self):
+        executed_by_url = self.cleaned_data['executed_by_url']
+        if executed_by_url:
+            # check that it's an active url
+            url = Request(executed_by_url)
+            try:
+                urlopen(url)
+            except (URLError, HTTPError):
+                self.add_error("executed_by_url", "Provided URL seems like a broken URL")
+        return executed_by_url
+
+    def update_metadata(self, metadata):
+        executed_by = self.cleaned_data['executed_by']
+        executed_by_url = self.cleaned_data['executed_by_url']
+        if executed_by:
+            metadata.executed_by = executed_by
+            metadata.executed_by_url = None
+        elif executed_by_url:
+            metadata.executed_by = None
+            metadata.executed_by_url = executed_by_url
+        else:
+            metadata.executed_by = None
+            metadata.executed_by_url = None
+
+        metadata.has_model_output = self.cleaned_data['has_model_output']
+        metadata.is_dirty = True
+        metadata.save()
+
+
+class ModelProgramMetadataValidationForm(forms.Form):
+    version = forms.CharField(required=False, max_length=250)
+    release_date = forms.DateField(required=False)
+    website = forms.URLField(required=False, max_length=255)
+    code_repository = forms.URLField(required=False, max_length=255)
+    programming_languages = forms.CharField(required=False)
+    operating_systems = forms.CharField(required=False)
+    mi_json_schema = forms.CharField(required=False)
```

Comment:
why isn't this called a model program schema?  I don't have a whole picture view of this yet, but I'd expect a model program schema here, not an instance of a model program.